keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -96,7 +100,7 @@ public static Nd4jDoubleTensor arange(double start, double end, double stepSize)
     }
 
     static INDArray unsafeGetNd4J(DoubleTensor that) {
-        if (that.isLengthOne()) {
+        if (that.isScalar() || that instanceof ScalarDoubleTensor) {
```

Comment:
I think there are some cases where a tensor is scalar but is not an instance of ScalarDoubleTensor or ScalarIntegerTensor. If that should never be the case, I can investigate where they come from and fix it (I think it can happen as a result of some operations).

Reply:
I think the check should still be for isLengthOne(). That's identifies the special case completely and doesn't couple the Nd4jDoubleTensor to the ScalarDoubleTensor.

Reply:
Changed it to `isLengthOne()`, good point.